api/client/client.go

```diff
@@ -3,3 +3,74 @@
 // Run "docker help SUBCOMMAND" or "docker SUBCOMMAND --help" to see more information on any Docker subcommand, including the full list of options supported for the subcommand.
 // See https://docs.docker.com/installation/ for instructions on installing Docker.
 package client
+
+import (
+	"io"
+
+	"github.com/docker/docker/api/client/lib"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/cliconfig"
+	"github.com/docker/docker/pkg/parsers/filters"
+	"github.com/docker/docker/registry"
+	"github.com/docker/docker/runconfig"
+)
+
+// apiClient is an interface that clients that talk with a docker server must implement.
+type apiClient interface {
+	ContainerAttach(options types.ContainerAttachOptions) (*types.HijackedResponse, error)
+	ContainerCommit(options types.ContainerCommitOptions) (types.ContainerCommitResponse, error)
+	ContainerCreate(config *runconfig.ContainerConfigWrapper, containerName string) (types.ContainerCreateResponse, error)
+	ContainerDiff(containerID string) ([]types.ContainerChange, error)
+	ContainerExecAttach(execID string, config *runconfig.ExecConfig) (*types.HijackedResponse, error)
+	ContainerExecCreate(config *runconfig.ExecConfig) (types.ContainerExecCreateResponse, error)
+	ContainerExecInspect(execID string) (types.ContainerExecInspect, error)
+	ContainerExecResize(execID string, height, width int) error
+	ContainerExecStart(execID string, config *types.ExecStartCheck) error
+	ContainerExport(containerID string) (io.ReadCloser, error)
```

Comment:
I think they should be only replaced if they have the general meaning of `any-ref-to-container` or `any-ref-to-image`. For example, if create/commit return only IDs they should keep the `ID` suffix. Can `execID` be anything else than ID?